### PR TITLE
update notebooks to reflect change in opentopo

### PR DESCRIPTION
**Pull request by bbuzz31**

### CODE REVIEW

**Comment by dbekaert** [comment]
@sssangha Can you review?

@bbuzz31 Could you add the capability to do notebook review? Perhaps check with @yunjunz or mintpy repo in case of questions how to enable it.

**Comment by yunjunz** [comment]
> @bbuzz31 Could you add the capability to do notebook review? Perhaps check with @yunjunz or mintpy repo in case of questions how to enable it.

It should be straightforward to link ReviewNB (https://www.reviewnb.com/) with the Github repo, one might need the admin permission to enable it.

**Comment by dbekaert** [comment]
Thanks @yunjunz I added it in now. Hopefully works for the next PR.

File: JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb [hunk 1]
```diff
@@ -1,10 +1,5 @@
 {
```
Comment (dbekaert) [inline JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>ITs not clear what API key for what.</p><ul><li>Explain first also why you need a key for what you need it and then how to get it.</li></ul><p> </p><p> </p><p>Also you just need it period, so does not add much information.</p><ul><li>Remove on same computer as running this notebook as cloud is not a computer.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (dbekaert) [inline JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>Do you need to keep the " " and put key on between or not? clarify</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (dbekaert) [inline JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>There is no plot in your notebook. Please check if this is an error and needs fix in code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (dbekaert) [inline JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>same no plot</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (dbekaert) [inline JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>same. think you did not run cells, but would be good to check if this still works.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

**Comment by dbekaert** [comment]
@bbuzz31 the rendering is not yet working, so best is to click on respond via the review-notebook tool. I did my review using the tool outside github as it was not showing here (I believe next time we will have it working).

In summary:
- I think some cells did not work, so would check if this is an error or not
- the verbose does not print product download summary
- the download does not give progress which could be problematic
- the version of ARIA-tools did we capture that now correctly in the code? @yunjunz had suggest a specific approach similar to mintpy where you pull it from some sort of version file. Suggest we adopt that.
- some more description would be beneficial for the program help function.

File: JupyterDocs/ariaDownload/ariaDownload_tutorial.ipynb [hunk 2]
```diff
@@ -54,9 +54,11 @@
      "name": "stdout",
```
[the remark this answers is not in the document]
Comment (bbuzz31) [reply JupyterDocs/ariaDownload/ariaDownload_tutorial.ipynb:54]: <p>There was a bug (fixed) but also asf_search cannot currently download KMZ/KML. I removed these cells from the notebook.</p>

[the remark this answers is not in the document]
Comment (bbuzz31) [reply JupyterDocs/ariaDownload/ariaDownload_tutorial.ipynb:54]: <p>Nope that's a bug. Fixed.</p>

[the remark this answers is not in the document]
Comment (bbuzz31) [reply JupyterDocs/ariaDownload/ariaDownload_tutorial.ipynb:54]: <p>Good question: how's this updated help string: "Specify version as str, e.g. 2_0_4 or all prods; default: 'newest. All products are downloaded. Unspecified versions are stored in "workdir"/duplicated_products'.</p>

File: JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb (hunk 1 above)
Comment (bbuzz31) [reply JupyterDocs/ariaMisclosure/ariaMisclosure_tutorial.ipynb:1]: <p>Hmm no you can just do <code>echo KEY &gt; ~/.topoapi</code></p><p>Is this what you meant? </p>

**Comment by bbuzz31** [comment]
<p>Per Sim: I believe so, but it prioritizes the new tif formats for sure</p><p>So it looks for duplicates and discards binary format if tif exists.</p><p> </p><p>Help is updated to reflect this.</p><p> </p><p> </p>

---

 <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bbuzz31** [comment]
<p>I had an older version of asf_search.py installed that errors out on duplicate products. New version just puts a warning.</p>

---

 <a href='https://app.reviewnb.com/aria-tools/ARIA-tools-docs/pull/46/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bbuzz31** [comment]
@dbekaert You caught a lot of bugs, thanks for going through these in such detail! 

I think I've addressed all the comments, and commits are here and  https://github.com/aria-tools/ARIA-tools/pull/306.
The one outstanding task is the progress bar for the download.

> @bbuzz31 the rendering is not yet working, so best is to click on respond via the review-notebook tool. I did my review using the tool outside github as it was not showing here (I believe next time we will have it working).
> 
> In summary:
> 
>     * I think some cells did not work, so would check if this is an error or not
> 
>     * the verbose does not print product download summary
> 
>     * the download does not give progress which could be problematic
> 
>     * the version of ARIA-tools did we capture that now correctly in the code? @yunjunz had suggest a specific approach similar to mintpy where you pull it from some sort of version file. Suggest we adopt that.
> 
>     * some more description would be beneficial for the program help function.

**Review by sssangha — APPROVED** [review]

**Comment by sssangha** [comment]
@bbuzz31 apologies, but this had slipped me. However, I had reshuffled the .netrc/.topoapi generation step to the top "prep" sections and made it so that a user explicitly has to enter their credentials if these files don't exist. Please refer to PR #47

**Comment by sssangha** [comment]
@bbuzz31 following my initial message, I believe it would be easier to merge my suggested changes here. Namely, I've added the following:
1. Add topoapi check + cleanup .netrc checks and move all to prep stage(s) at top of notebook.
2. If a given file doesn't exist, and explicit warning is passed, plus a prompt for users to enter their credentials.

I'll keep the aforementioned PR in draft form for now for reference, but I intend to merge this to master

**Comment by bbuzz31** [comment]
@sssangha there are numerous other small changes in this PR besides the topo check. Can you clarify whether you're going to merge this branch and #47?

**Comment by sssangha** [comment]
@bbuzz31 my plan is to close #47 and merge this one. I had essentially moved the .netrc and .topoapi checks to the top within the Prep sections. We could merge and proceed with this branch

**Comment by bbuzz31** [comment]
Oh I see now. Sounds good!

**Pull request merged**